webapp/src/components/issue_attribute_selector.jsx

```diff
@@ -0,0 +1,105 @@
+
+export default class IssueAttributeSelector extends PureComponent {
+    static propTypes = {
+        isMulti: PropTypes.bool.isRequired,
+        repo: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        onChange: PropTypes.func.isRequired,
+        loadOptions: PropTypes.func.isRequired,
+        selection: PropTypes.oneOfType([
+            PropTypes.object,
+            PropTypes.array,
+            PropTypes.string,
+        ]).isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            options: [],
+            isLoading: false,
+            error: null,
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.repo && prevProps.repo !== this.props.repo) {
+            this.loadOptions();
+        }
+    }
+
+    loadOptions = async () => {
+        this.setState({isLoading: true});
+
+        try {
+            const options = await this.props.loadOptions();
+            this.filterSelection(options);
+            this.setState({
+                options,
+                isLoading: false,
+            });
+        } catch (err) {
+            this.setState({
+                error: err,
+                isLoading: false,
+            });
+        }
+    };
+
+    filterSelection = (options) => {
+        const optionValues = options.map((option) => option.value);
+
+        // filter out currently selected options that do not exist in the new repo
+        let validSelection;
+        if (this.props.isMulti) {
+            validSelection = this.props.selection.filter((v) => optionValues.includes(v.value));
+        } else {
+            validSelection = optionValues.includes(this.props.selection.value) ? this.props.selection : {};
+        }
+
+        this.props.onChange(validSelection);
+    }
+
+    onChange = (selection) => {
+        const blank = this.props.isMulti ? [] : '';
+        this.props.onChange(selection || blank);
+    };
+
+    render() {
+        return (
+            <Setting {...this.props}>
+                <ReactSelect
+                    {...this.props}
+                    placeholder={'Select...'}
+                    closeMenuOnSelect={!this.props.isMulti}
+                    hideSelectedOptions={this.props.isMulti}
+                    onChange={this.onChange}
+                    options={this.state.options}
+                    value={this.props.selection}
+                    isLoading={this.state.isLoading}
+                    styles={getStyleForReactSelect(this.props.theme)}
+                />
+
+                {this.state.error && (
+                    <p className='alert alert-danger'>
+                        <i
+                            className='fa fa-warning'
+                            title='Warning Icon'
+                        />
+                        <span> {this.state.error.message}</span>
+                    </p>
+                )}
```

Comment:
I tried to test this code by modifying all API endpoints to return 500.

Should an error icon be shown in this case?
![Screenshot from 2020-08-06 11-48-53](https://user-images.githubusercontent.com/16541325/89518178-d71bff00-d7da-11ea-8e6c-cf112ced8df0.png)

Reply:
@hanzei Very good catch, I fixed it. This is how it looks now in case of a server error:

![image](https://user-images.githubusercontent.com/15212283/89560901-6b5c8500-d820-11ea-9a04-344bcf017a64.png)

Reply:
Thanks for fixing this issue :+1: 

I've found a minor edge case while testing:
1. Modify all endpoints to return 500 and deploy
3. Open "Create issue dialog"
4. Select a repository
5. Confirm that error message is shown
6. Revert changes to endpoint and deploy to simulate a partial outage
7. Select another repository

Expected:
Error message disappears once new repository is selected. Labels, Assignees and Milestones can be correctly selected.

Observed:
Error message persists once new repository is selected. Labels, Assignees and Milestones can be correctly selected.


This is totally an edge case and not a blocker for this PR. Feel free to ignore it, if it's to hard to fix.

Reply:
@hanzei Thanks for pointing that out, I fixed the bug :+1: